### PR TITLE
Use GHC 8.0 as the minimum

### PR DESCRIPTION
**Pull request by RyanGlScott**
In the spirit of ekmett/lens#977.

### CODE REVIEW

File: trifecta.cabal [hunk 1]
```diff
@@ -82,7 +80,7 @@ library
     ghc-prim,
     hashable                    >= 1.2.1   && < 1.4,
     indexed-traversable         >= 0.1.1   && < 0.2,
-    lens                        >= 4.0     && < 6,
+    lens                        >= 4.13    && < 6,
```
Comment (phadej) [inline trifecta.cabal:83]: Why `4.13`. Is there a reason to not crank it up higher, to `4.19.2` or even `5`?

Comment (RyanGlScott) [reply trifecta.cabal:83]: No reason other than I wasn't sure what the most recent version bounds I could reasonably support are. At the very least, I needed `4.13` to remove the CPP [here](https://github.com/ekmett/trifecta/blob/37cd03a4ab249449645b344ee8fc83959153679f/src/Text/Trifecta/Highlight.hs#L29), so I opted to be conservative and make that the minimum. Perhaps someone more clever than I could do an analysis of all possible version bounds to figure out what the most recent version we could support is.

Comment (phadej) [reply trifecta.cabal:83]: The removal of CPP makes the bound justified: it's not an arbitrary. Let me do a quick test before this is merged though.

**Comment by phadej** [comment]
Re bounds: bounds on boot libs can be tightened, GHC-8.0.2 comes with:
- containers-0.5.7.1
- bytestring-0.10.8.1
- transformers-0.5.2.0
- mtl-2..2.2
- array-0.5.1.1
- deepseq-1.4.2.0

There doesn't seem to be any CPP related to them though.

**Comment by phadej** [comment]
```
                             8.0.2    8.2.2     8.4.4     
ansi-terminal                0.6      0.6       0.6       
array                        0.5.1.1  0.5.2.0   0.5.2.0     -- here (boot)
base                         4.9.1.0  4.10.1.0  4.11.1.0  
blaze-builder                0.3.3.4  0.3.3.4   0.4.1.0   
blaze-html                   0.9.0.0  0.9.0.0   0.9.0.0     -- here
blaze-markup                 0.8.0.0  0.8.0.0   0.8.1.0   
bytestring                   0.9.2.1  0.9.2.1   0.10.8.0  
charset                      0.3.7    0.3.7     0.3.7       
comonad                      5        5.0.1     5.0.1       -- here
containers                   0.5.7.0  0.5.7.0   0.5.7.0     -- here (boot)
deepseq                      1.4.2.0  1.4.3.0   1.4.3.0     -- here (boot)
fingertree                   0.1.0.0  0.1.0.0   0.1.2.0     -- here
ghc-prim                     0.5.0.0  0.5.1.1   0.5.2.0     
hashable                     1.2.4.0  1.2.6.1   1.2.7.0     -- (i'd suggest tightening to 1.2.7.0 anyway)
indexed-traversable          0.1.1    0.1.1     0.1.1       
lens                         4.14     4.15.2    4.16        -- here
mtl                          2.2.1    2.2.1     2.2.1       -- here (almost boot)
parsers                      0.12.4   0.12.5    0.12.5    
prettyprinter                1.7.0    1.7.0     1.7.0     
prettyprinter-ansi-terminal  1.1.2    1.1.2     1.1.2     
profunctors                  5.2      5.2       5.2.2       -- here
reducers                     3.12.1   3.12.1    3.12.2      -- here
transformers                 0.5.0.0  0.5.0.0   0.5.5.0     -- here (boot)
unordered-containers         0.2.6.0  0.2.6.0   0.2.9.0   
utf8-string                  0.3.6    0.3.6     0.3.6     
```

I run a lower-bounds check, and there're results. I marked where we can tighten the lower bounds.

**Comment by RyanGlScott** [comment]
Thanks, @phadej! Out of curiosity, are you using a tool of some kind to produce these results? Having the ability to do that myself might be useful in the future.

**Comment by phadej** [comment]
@RyanGlScott https://github.com/phadej/trustee, but I have to warn that it comes *as is*.

**Pull request merged**